qiskit/transpiler/layout.py

```diff
@@ -370,7 +370,35 @@ def from_qubit_list(qubit_list, *qregs):
 
 @dataclass
 class TranspileLayout:
-    """Layout attributes from output circuit from transpiler."""
+    r"""Layout attributes from output circuit from transpiler.
+
+    The transpiler in general is unitary-perserving up to permutations caused
+    by setting and applying initial layout during the :ref:`layout_stage`
+    and :class:`~.SwapGate` insertion during the :ref:`routing_stage`. To
+    provide an interface to reason about these permutations caused by
+    the :mod:`~qiskit.transpiler`.
+
+    There are three attributes associated with the class:
+
+      * :attr:`initial_layout` - This attribute is used to model the
+        permutation caused by the :ref:`layout_stage` it contains a
+        :class:`~.Layout` object that maps the input :class:`~.QuantumCircuit`\s
+        :class:`~.Qubit` objects to the position in the output
+        :class:`.QuantumCircuit.qubits` list.
```

Comment:
This layout includes auto-inserted ancillas under the implicitly assumed default `QuantumRegister(QuantumRegister(backend.num_qubits - circuit.num_qubits, "ancilla")`, "ancilla")`

Reply:
This is something that needs to be addressed probably with an additional field being added since it's a bit more involved to track this. There is an issue open about this here: https://github.com/Qiskit/qiskit-terra/issues/9476